panic conversion

```
± |master ✓| → c2go transpile main.c frame3dd.c frame3dd_io.c coordtrans.c eig.c HPGmatrix.c HPGutil.c NRutil.c 
panic: interface conversion: ast.Node is *ast.BinaryOperator, not *ast.CompoundStmt

goroutine 1 [running]:
github.com/elliotchance/c2go/transpiler.transpileForStmt(0xc4219ca960, 0xc4200961a0, 0xc4220f7200, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/branch.go:218 +0x2769
github.com/elliotchance/c2go/transpiler.transpileToStmt(0x6c69e0, 0xc4219ca960, 0xc4200961a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:339 +0x6fb
github.com/elliotchance/c2go/transpiler.transpileToStmts(0x6c69e0, 0xc4219ca960, 0xc4200961a0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:276 +0x105
github.com/elliotchance/c2go/transpiler.transpileCompoundStmt(0xc42067e000, 0xc4200961a0, 0xc422950ff0, 0xc422951008, 0x6c64e0, 0xc42067e000, 0x7, 0x0, 0x427101, 0x0, ...)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/scope.go:21 +0x10d
github.com/elliotchance/c2go/transpiler.transpileToStmt(0x6c64e0, 0xc42067e000, 0xc4200961a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:345 +0xc5f
github.com/elliotchance/c2go/transpiler.transpileToStmts(0x6c64e0, 0xc42067e000, 0xc4200961a0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:276 +0x105
github.com/elliotchance/c2go/transpiler.transpileToBlockStmt(0x6c64e0, 0xc42067e000, 0xc4200961a0, 0x0, 0x1, 0xc420f39f60, 0x0, 0x1, 0x0, 0x0, ...)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/scope.go:38 +0x46
github.com/elliotchance/c2go/transpiler.transpileFunctionDecl(0xc4209de4e0, 0xc4200961a0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/functions.go:85 +0x1b11
github.com/elliotchance/c2go/transpiler.transpileToNode(0x6c6a60, 0xc4209de4e0, 0xc4200961a0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:433 +0x511
github.com/elliotchance/c2go/transpiler.transpileTranslationUnitDecl(0xc4200961a0, 0xc4219c82a0, 0xc421f79768, 0xc4200961a0, 0x6c73a0, 0xc4219c82a0, 0xc421f79768)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/translation_unit.go:48 +0x7ec
github.com/elliotchance/c2go/transpiler.transpileToNode(0x6c73a0, 0xc4219c82a0, 0xc4200961a0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:430 +0x65e
github.com/elliotchance/c2go/transpiler.TranspileAST(0x0, 0x0, 0x5e67a7, 0x4, 0xc4200961a0, 0x6c73a0, 0xc4219c82a0, 0x2, 0x56000)
	/home/lepricon/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:32 +0x202
main.Start(0x0, 0xc42008e020, 0x8, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5e67a7, ...)
	/home/lepricon/go/src/github.com/elliotchance/c2go/main.go:294 +0xdc8
main.runCommand(0xc420072000)
	/home/lepricon/go/src/github.com/elliotchance/c2go/main.go:419 +0x4af
main.main()
	/home/lepricon/go/src/github.com/elliotchance/c2go/main.go:343 +0x22
```

minimal C example:
```c
void v()
{
        long i,j;
        for(i=0,j=0;i<=4;i++,j++) i+=j;
        return;
}
int main(){
        return 0;
}
```
Error:
```
panic: interface conversion: ast.Node is *ast.CompoundAssignOperator, not *ast.CompoundStmt

goroutine 1 [running]:
github.com/elliotchance/c2go/transpiler.transpileForStmt(0xc4200cef60, 0xc4200be000, 0xc4200c9780, 0xc4202cd200, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, ...)
	/home/konstantin/go/src/github.com/elliotchance/c2go/transpiler/branch.go:224 +0x2017
github.com/elliotchance/c2go/transpiler.transpileToStmt(0x6b1c60, 0xc4200cef60, 0xc4200be000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/konstantin/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:367 +0x85b
github.com/elliotchance/c2go/transpiler.transpileToStmts(0x6b1c60, 0xc4200cef60, 0xc4200be000, 0x0, 0x0, 0x0, 0x0, 0x0)
```

AST tree:
```
|-FunctionDecl 0x1e544c8 <1.c:1:1, line:6:1> line:1:6 v 'void ()'
| `-CompoundStmt 0x1ea0080 <line:2:1, line:6:1>
|   |-DeclStmt 0x1e546a8 <line:3:9, col:17>
|   | |-VarDecl 0x1e545c0 <col:9, col:14> col:14 used i 'long'
|   | `-VarDecl 0x1e54630 <col:9, col:16> col:16 used j 'long'
|   |-ForStmt 0x1ea0030 <line:4:9, col:38>
|   | |-BinaryOperator 0x1e547d0 <col:13, col:19> 'long' ','
|   | | |-BinaryOperator 0x1e54720 <col:13, col:15> 'long' '='
|   | | | |-DeclRefExpr 0x1e546c0 <col:13> 'long' lvalue Var 0x1e545c0 'i' 'long'
|   | | | `-ImplicitCastExpr 0x1e54708 <col:15> 'long' <IntegralCast>
|   | | |   `-IntegerLiteral 0x1e546e8 <col:15> 'int' 0
|   | | `-BinaryOperator 0x1e547a8 <col:17, col:19> 'long' '='
|   | |   |-DeclRefExpr 0x1e54748 <col:17> 'long' lvalue Var 0x1e54630 'j' 'long'
|   | |   `-ImplicitCastExpr 0x1e54790 <col:19> 'long' <IntegralCast>
|   | |     `-IntegerLiteral 0x1e54770 <col:19> 'int' 0
|   | |-<<<NULL>>>
|   | |-BinaryOperator 0x1e54870 <col:21, col:24> 'int' '<='
|   | | |-ImplicitCastExpr 0x1e54840 <col:21> 'long' <LValueToRValue>
|   | | | `-DeclRefExpr 0x1e547f8 <col:21> 'long' lvalue Var 0x1e545c0 'i' 'long'
|   | | `-ImplicitCastExpr 0x1e54858 <col:24> 'long' <IntegralCast>
|   | |   `-IntegerLiteral 0x1e54820 <col:24> 'int' 4
|   | |-BinaryOperator 0x1e9ff68 <col:26, col:31> 'long' ','
|   | | |-UnaryOperator 0x1e9ff00 <col:26, col:27> 'long' postfix '++'
|   | | | `-DeclRefExpr 0x1e54898 <col:26> 'long' lvalue Var 0x1e545c0 'i' 'long'
|   | | `-UnaryOperator 0x1e9ff48 <col:30, col:31> 'long' postfix '++'
|   | |   `-DeclRefExpr 0x1e9ff20 <col:30> 'long' lvalue Var 0x1e54630 'j' 'long'
|   | `-CompoundAssignOperator 0x1e9fff8 <col:35, col:38> 'long' '+=' ComputeLHSTy='long' ComputeResultTy='long'
|   |   |-DeclRefExpr 0x1e9ff90 <col:35> 'long' lvalue Var 0x1e545c0 'i' 'long'
|   |   `-ImplicitCastExpr 0x1e9ffe0 <col:38> 'long' <LValueToRValue>
|   |     `-DeclRefExpr 0x1e9ffb8 <col:38> 'long' lvalue Var 0x1e54630 'j' 'long'
|   `-ReturnStmt 0x1ea0068 <line:5:9>
```

C example:
```c
void v()
{
        long i,j;
        for(i=0,j=0;i<=4;i++,j++) i=i+j;
        return;
}
int main(){
        return 0;
}
```
Error:
```
panic: interface conversion: ast.Node is *ast.BinaryOperator, not *ast.CompoundStmt

goroutine 1 [running]:
github.com/elliotchance/c2go/transpiler.transpileForStmt(0xc4200c65a0, 0xc4200cc4e0, 0xc4200b7f00, 0xc420312040, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, ...)
	/home/konstantin/go/src/github.com/elliotchance/c2go/transpiler/branch.go:224 +0x2017
github.com/elliotchance/c2go/transpiler.transpileToStmt(0x6b1c60, 0xc4200c65a0, 0xc4200cc4e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/konstantin/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:367 +0x85b
github.com/elliotchance/c2go/transpiler.transpileToStmts(0x6b1c60, 0xc4200c65a0, 0xc4200cc4e0, 0x0, 0x0, 0x0, 0x0, 0x0)
```
AST tree:
```
|-FunctionDecl 0x38cb4c8 <1.c:1:1, line:6:1> line:1:6 v 'void ()'
| `-CompoundStmt 0x39170d8 <line:2:1, line:6:1>
|   |-DeclStmt 0x38cb6a8 <line:3:9, col:17>
|   | |-VarDecl 0x38cb5c0 <col:9, col:14> col:14 used i 'long'
|   | `-VarDecl 0x38cb630 <col:9, col:16> col:16 used j 'long'
|   |-ForStmt 0x3917088 <line:4:9, col:39>
|   | |-BinaryOperator 0x38cb7d0 <col:13, col:19> 'long' ','
|   | | |-BinaryOperator 0x38cb720 <col:13, col:15> 'long' '='
|   | | | |-DeclRefExpr 0x38cb6c0 <col:13> 'long' lvalue Var 0x38cb5c0 'i' 'long'
|   | | | `-ImplicitCastExpr 0x38cb708 <col:15> 'long' <IntegralCast>
|   | | |   `-IntegerLiteral 0x38cb6e8 <col:15> 'int' 0
|   | | `-BinaryOperator 0x38cb7a8 <col:17, col:19> 'long' '='
|   | |   |-DeclRefExpr 0x38cb748 <col:17> 'long' lvalue Var 0x38cb630 'j' 'long'
|   | |   `-ImplicitCastExpr 0x38cb790 <col:19> 'long' <IntegralCast>
|   | |     `-IntegerLiteral 0x38cb770 <col:19> 'int' 0
|   | |-<<<NULL>>>
|   | |-BinaryOperator 0x38cb870 <col:21, col:24> 'int' '<='
|   | | |-ImplicitCastExpr 0x38cb840 <col:21> 'long' <LValueToRValue>
|   | | | `-DeclRefExpr 0x38cb7f8 <col:21> 'long' lvalue Var 0x38cb5c0 'i' 'long'
|   | | `-ImplicitCastExpr 0x38cb858 <col:24> 'long' <IntegralCast>
|   | |   `-IntegerLiteral 0x38cb820 <col:24> 'int' 4
|   | |-BinaryOperator 0x3916f68 <col:26, col:31> 'long' ','
|   | | |-UnaryOperator 0x3916f00 <col:26, col:27> 'long' postfix '++'
|   | | | `-DeclRefExpr 0x38cb898 <col:26> 'long' lvalue Var 0x38cb5c0 'i' 'long'
|   | | `-UnaryOperator 0x3916f48 <col:30, col:31> 'long' postfix '++'
|   | |   `-DeclRefExpr 0x3916f20 <col:30> 'long' lvalue Var 0x38cb630 'j' 'long'
|   | `-BinaryOperator 0x3917060 <col:35, col:39> 'long' '='
|   |   |-DeclRefExpr 0x3916f90 <col:35> 'long' lvalue Var 0x38cb5c0 'i' 'long'
|   |   `-BinaryOperator 0x3917038 <col:37, col:39> 'long' '+'
|   |     |-ImplicitCastExpr 0x3917008 <col:37> 'long' <LValueToRValue>
|   |     | `-DeclRefExpr 0x3916fb8 <col:37> 'long' lvalue Var 0x38cb5c0 'i' 'long'
|   |     `-ImplicitCastExpr 0x3917020 <col:39> 'long' <LValueToRValue>
|   |       `-DeclRefExpr 0x3916fe0 <col:39> 'long' lvalue Var 0x38cb630 'j' 'long'
|   `-ReturnStmt 0x39170c0 <line:5:9>
```

--- a/tests/for.c
+++ b/tests/for.c
@@ -3,7 +3,7 @@
 
 int main()
 {
-    plan(50);
+    plan(53);
 
     int i = 0;
 
@@ -119,6 +119,26 @@ int main()
 		}
 	}
 	is_eq( nn2 , 3 );
-    
+
+	diag("for with specific body");
+	{
+	        double vars = 42.0;
+	        for(int s = 0;s<1;s++)
+	                vars ++;
+	        is_eq(vars,43.0);
+	}
+	{
+	        double vars = 42.0;
+	        for(int s = 0;s<1;s++)
+	                vars += 1.0;
+	        is_eq(vars,43.0);
+	}
+	{
+	        double vars = 42.0;
+	        for(int s = 0;s<1;s++)
+	                vars = vars + 1.0;
+	        is_eq(vars,43.0);
+	}
+
 	done_testing();
 }

--- a/tests/for.c
+++ b/tests/for.c
@@ -3,7 +3,7 @@
 
 int main()
 {
-    plan(53);
+    plan(57);
 
     int i = 0;
 
@@ -135,9 +135,35 @@ int main()
 	}
 	{
 	        double vars = 42.0;
+			int X,Y;
+	        for(X = 0, Y = 0; X < 1;X++,Y++)
+	                vars += 1.0;
+	        is_eq(vars,43.0);
+	}
+	{
+	        double vars = 42.0;
 	        for(int s = 0;s<1;s++)
 	                vars = vars + 1.0;
 	        is_eq(vars,43.0);
+	}
+	{
+	        double vars = 42.0;
+			int X,Y;
+	        for(X = 0, Y = 0; X < 1;X++,Y++)
+	                vars = vars + 1.0;
+	        is_eq(vars,43.0);
+	}
+	{
+			int X,Y;
+	        for(X = 0, Y = 0; X < 1;X++,Y++)
+	                X += Y;
+	        pass("ok");
+	}
+	{
+			int X,Y;
+	        for(X = 0, Y = 0; X < 1;X++,Y++)
+	                X = X + Y;
+	        pass("ok");
 	}
 
 	done_testing();

--- a/transpiler/branch.go
+++ b/transpiler/branch.go
@@ -217,6 +217,7 @@ func transpileForStmt(n *ast.ForStmt, p *program.Program) (
 				// if body is exist
 				if _, ok := children[4].(*ast.CompoundStmt); !ok {
 					compound = new(ast.CompoundStmt)
+					compound.AddChild(children[4])
 				} else {
 					compound = children[4].(*ast.CompoundStmt)
 				}

--- a/transpiler/branch.go
+++ b/transpiler/branch.go
@@ -215,7 +215,11 @@ func transpileForStmt(n *ast.ForStmt, p *program.Program) (
 			var compound *ast.CompoundStmt
 			if children[4] != nil {
 				// if body is exist
-				compound = children[4].(*ast.CompoundStmt)
+				if _, ok := children[4].(*ast.CompoundStmt); !ok {
+					compound = new(ast.CompoundStmt)
+				} else {
+					compound = children[4].(*ast.CompoundStmt)
+				}
 			} else {
 				// if body is not exist
 				compound = new(ast.CompoundStmt)

